packages/core/src/lib/streamManager/index.ts

```diff
@@ -32,11 +34,33 @@ import { ToolHandler } from './clientTools/handlers'
 import { ResolvedTools } from './resolveTools/types'
 import { createPromiseWithResolver } from './utils/createPromiseResolver'
 
-const EMPTY_USAGE = (): LanguageModelUsage => ({
-  promptTokens: 0,
-  completionTokens: 0,
-  totalTokens: 0,
-})
+const addTokens = ({
+  attr,
+  prev,
+  next,
+}: {
+  attr: keyof LanguageModelUsage
+  prev: LanguageModelUsage | undefined
+  next: LanguageModelUsage
+}) => prev?.[attr] ?? 0 + next[attr]
```

Comment:
```suggestion
}) => (prev?.[attr] ?? 0) + next[attr]
```

put some parenthesis in there just in case it mistankenly does only prev or either 0+next